google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/PollingSubscriberConnection.java

```diff
@@ -128,37 +128,17 @@ protected void doStop() {
 
   private void pullMessages(final Duration backoff) {
     ListenableFuture<PullResponse> pullResult =
-        stub.withDeadlineAfter(DEFAULT_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS)
-            .pull(
-                PullRequest.newBuilder()
-                    .setSubscription(subscription)
-                    .setMaxMessages(maxDesiredPulledMessages)
-                    .setReturnImmediately(true)
-                    .build());
+        stub.pull(
+            PullRequest.newBuilder()
+                .setSubscription(subscription)
+                .setMaxMessages(maxDesiredPulledMessages)
+                .build());
 
     Futures.addCallback(
         pullResult,
         new FutureCallback<PullResponse>() {
           @Override
           public void onSuccess(PullResponse pullResponse) {
-            if (pullResponse.getReceivedMessagesCount() == 0) {
```

Comment:
you actually want to keep this, even with setReturnImmediately(false) for scalability reasons we terminate long outstanding pulls with an empty messages response.